### PR TITLE
fix: bump aws-sdk-go-base to v2.0.0-beta.24

### LINKED ISSUE #33349: [Bug]: s3 based terraform_remote_state not support ap-south-2 region
### Terraform Core Version

1.4.6

### AWS Provider Version

5.0.1

### Affected Resource(s)

* terraform_remote_state

### Expected Behavior

Terraform should be able to read s3 based remote state in ap-south-2 region

### Actual Behavior

Terraform plan return invalid region error

### Relevant Error/Panic Output Snippet

```shell
❯ terraform plan
data.terraform_remote_state.test: Reading...

Planning failed. Terraform encountered an error while generating this plan.
╷
│ Error: Invalid AWS Region: ap-south-2

```

### Terraform Configuration Files

```terraform
provider "aws" {
  profile = "india-hyd"
  region  = "ap-south-2"
}

data "terraform_remote_state" "test" {
  backend = "s3"
  config = {
    bucket = "test"
    key    = "test.tfstate"
  }
}
```

### Steps to Reproduce

Run terraform plan with config above

It working if I add
```
skip_region_validation = true
```

### Debug Output

_No response_

### Panic Output

_No response_

### Important Factoids

_No response_

### References

_No response_

### Would you like to implement a fix?

None

### PR DESCRIPTION
**Pull request by yushao2**
This is a small chore for s3 backend, the pinned module version for the aws helper is outdated and does not contain many regions that are supported by AWS today. This change bumps it up to the latest release on  https://github.com/hashicorp/aws-sdk-go-base/releases

Fixes #33349

## Target Release

1.5.x

### BUG FIXES

- Terraform will no longer fail to read s3 based remote state in newer regions. 

 

-

### CODE REVIEW

**Comment by yushao2** [comment]
A little more involved that what I thought it would be, but seems like I finally managed to get all tests working. Ready for review

File: internal/backend/remote-state/s3/backend.go [hunk 1]
```diff
@@ -164,13 +166,6 @@ func New() backend.Backend {
 				Default:     false,
 			},
 
-			"skip_metadata_api_check": {
-				Type:        schema.TypeBool,
-				Optional:    true,
-				Description: "Skip the AWS Metadata API check.",
-				Default:     false,
-			},
-
```
Comment (yushao2) [inline internal/backend/remote-state/s3/backend.go:173]: I removed this, as it seems to be irrelevant as far as configuring a s3 backend goes.

Comment (kmoe) [reply internal/backend/remote-state/s3/backend.go:173]: I'm not @hashicorp/terraform-aws, but it's worth noting that changing the schema for backends can cause users supplying backend configuration via variables to need to reconfigure the backend: https://github.com/hashicorp/terraform/issues/32866#issuecomment-1471921440
We therefore tend to be conservative in changing backend schemas.
